crates/hir_def/src/item_tree.rs

```diff
@@ -58,6 +58,13 @@ impl fmt::Debug for RawVisibilityId {
     }
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
```

Comment:
Is this is good design? (The name is questionable)

Reply:
The type could contain macro calls further inside (e.g. `Vec<macro!()>`), so I think you have to change `TypeRef` (or make a separate version that can contain macro calls)

Reply:
Yeah, I currently think the best way to do this is to add another variant to the existing `TypeRef`. It should then be replaced by the expanded type during lowering, or left as-is if the macro fails to resolve/expand.

Reply:
Oh crap. Yes, I guess this requires more work. Does it makes sense to just extend `TypeRef` to have a `MacroType` variant? However, I could imagine that "traversing" the `TypeRef` to see whether TypeRef contains a `MacroType` can be cumbersome.